src/gui/screen_menu_lan_settings.cpp

```diff
@@ -100,69 +99,67 @@ ScreenMenuLanSettings::ScreenMenuLanSettings()
     : AddSuperWindow<screen_t>(nullptr, win_type_t::normal, is_closed_on_timeout_t::no)
     , menu(this, GuiDefaults::RectScreenBody - Rect16::Height_t(get_help_h()), &container)
     , header(this)
-    , help(this, Rect16(GuiDefaults::RectScreen.Left(), uint16_t(GuiDefaults::RectScreen.Height()) - get_help_h(), GuiDefaults::RectScreen.Width(), get_help_h()), is_multiline::yes) {
+    , help(this, Rect16(GuiDefaults::RectScreen.Left(), uint16_t(GuiDefaults::RectScreen.Height()) - get_help_h(), GuiDefaults::RectScreen.Width(), get_help_h()), is_multiline::yes)
+    , cached_network_status(NETDEV_UNLINKED) {
     header.SetText(_(label));
     help.font = resource_font(helper_font);
     menu.GetActiveItem()->SetFocus(); // set focus on new item//containder was not valid during construction, have to set its index again
     CaptureNormalWindow(menu);        // set capture to list
-
-    refresh_addresses();
-    msg_shown = false;
 }
 
 /*****************************************************************************/
 //non static member function definition
-void ScreenMenuLanSettings::refresh_addresses() {
-    if (netdev_get_status(netdev_get_active_id()) == NETDEV_NETIF_UP) {
+void ScreenMenuLanSettings::refresh_addresses(uint32_t netdev_id) {
+    netdev_status_t status = netdev_get_status(netdev_id);
+    if (status == cached_network_status) {
+        return;
+    }
```

Comment:
Is this the right place to cache it? Because the current IP may be used by other parts of the code.

Also, this introduces a reverse problem ‒ the IP not being refreshed enough. I think this could happen in two cases:

* I've both wifi and ethernet configured, with signal, cable connected ‒ therefore both are connected, one of them is active. I switch from one to another, in both cases the state is `NETDEV_NETIF_UP`, so the IP does not get refreshed.
* I take one cable out, plug another one in, it changes the IP, goes down and up. This happens during one refresh cycle, so the down is not seen by the GUI at all, and shows the old IP.